src/main/java/by/exadel/internship/service/impl/FileServiceImpl.java

```diff
@@ -45,18 +52,54 @@ private String uploadFile(byte[] file, String fileName){
         log.info("Try to upload file to cloud");
         BlobId blobId = BlobId.of(bucketName, fileName);
         BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType("media").build();
-        Credentials credentials = null;
+
+        Credentials credentials = getCredentialsByJSON();
+
+        Storage storage = StorageOptions.newBuilder().setCredentials(credentials).build().getService();
+        storage.create(blobInfo, file);
+        log.info("File was uploaded to cloud");
+        return fileName;
+    }
+
+    @Override
+    public String download(String fileName, String formLastName) {
+        log.info("Try to download file from cloud");
+        String destFileName = formLastName
+                .concat(DOT_SEPARATOR)
+                .concat(this.getExtension(fileName));
+
+        Credentials credentials = getCredentialsByJSON();
+
+        Storage storage = StorageOptions.newBuilder().setCredentials(credentials).build().getService();
+        Blob blob = storage.get(BlobId.of(bucketName, fileName));
+
+        blob.downloadTo(Paths.get(destFileName));
```

Comment:
Why do you download file to local and after wrap it by ByteArrayResource ?

Reply:
Fixed